amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```diff
@@ -257,4 +266,15 @@ public ShutdownReason getReason() {
         return reason;
     }
 
+    private void dropLease() {
+        Lease currentLease = leaseCoordinator.getCurrentlyHeldLease(leaseKeyProvider.apply(shardInfo));
+        if (currentLease == null) {
+            log.warn("Shard " + shardInfo.shardId() + ": Lease already dropped. Will shutdown the shardConsumer directly.");
```

Comment:
We do not know if the lease was dropped. Just indicate unable to find the lease.

Reply:
Addressed